enhancements/olm-managed-metrics.md

```diff
@@ -0,0 +1,146 @@
+  name string
+
+  //+optional
+  // labels are labels that must be present on the given namespace.
+  labels []string
+}
+```
+
+>Note: If `NamespaceRequirements.namespace` is set, the UI will only show the specified namespace as an install option.
+
+#### ServiceMonitor and PrometheusRule Support
+
+As OLM gains support for these Prometheus objects, there is a chance that installed operators may overload the Prometheus instance. It is important that a Cluster Admin is alerted that an operator will report metrics prior to deploying the operator. As such, we propose adding a `monitoringSupported` boolean field to the `ClusterServiceVersion` object. OLM will not install either of the Prometheus object unless the `monitoringSupported` field is set to true. Additionally, Cluster Admins will need to manually the operator's `InstallPlans` in the following scenarios:
+
+- A fresh install for this operator is being created where this field is set to `true`
+- An existing operator had this field set to `false` and is being upgraded to a new version that sets this field to `true`
+
+> Note: The `InstallPlan` will include a helpful message for Cluster Admins if either of the situations above are encountered.
+
+### OLM Telemeter Support
+
+#### Telemeter Requirements
+
+Once OLM is capable of managing `ServiceMonitor` and `PrometheusRule` objects along with the lifecycle of an operator, OLM must allow Red Hat owned operators to take advantage of Telemeter. Before moving forward, let's review the requirements that must be met for Telemeter to scrape an application's metrics:
+
+1. The application must be deployed in a namespace that is both prefixed with `openshift-` and labbeled with `openshift.io/cluster-monitoring=true`
+2. A [ServiceMonitor object](https://github.com/coreos/prometheus-operator/blob/master/Documentation/user-guides/getting-started.md#related-resources) must be created in the namespace mentioned above and point to the application's metrics endpoint
+3. A role/rolebinding must be created that provides the Prometheus Operator ServiceAccount with the appropriate RBAC priviledges to discover the `ServiceMonitors` and `PrometheusRules` objects in the newly created namespace.
+
+Both requirements 1 and 2 are fully met by the changes proposed above leaving our team with the duty to fulfill requirement 3.
+
+#### Fulfilling Requirement 3: Updating the Prometheus Operator ServiceAccount
+
+In order for the Prometheus Operator to recognize take advantage of this new feature, its ServiceAccount must have the proper RBAC permissions to view events on the `ServiceMonitors` and `PrometheusRules` objects in the operator's namespace.
+
+On upstream clusters, there is no way for OLM to identify which Prometheus Operator ServiceAccount should be granted the RBAC priviledges. As such, the cluster admin will need to update the correct Prometheus Operator's OperatorGroup to include the correct namespace.
```

Comment:
Don't we just need to include the RoleBinding in the operator bundle?

Reply:
The operator author that builds the bundle wouldn't know which service account to bind the RBAC to either.